File: 1461-check-if-a-string-contains-all-binary-codes-of-size-k/1461-check-if-a-string-contains-all-binary-codes-of-size-k.py
class Solution:
    def hasAllCodes(self, s: str, k: int) -> bool:
        seen = set()
        n = len(s)
        for i in range(k-1, n):
            seen.add(s[i-k+1:i+1])
        return len(seen) == (2**k)

File: 1461-check-if-a-string-contains-all-binary-codes-of-size-k/1461-check-if-a-string-contains-all-binary-codes-of-size-k.cpp
class Solution {
public:
    bool hasAllCodes(string s, int k) {
        unordered_set<string> seen;
        int n = s.length();
        for (int i=(k-1); i<n; i++) {
            seen.insert(s.substr(i-k+1,k));
        }
        return (seen.size() == ((int) pow(2, k)));
    }
};